sdk/storage/azblob/internal/exported/blob_batch.go

```diff
@@ -0,0 +1,287 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package exported
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/uuid"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/shared"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type BlobBatchOperationType string
+
+const (
+	BatchIdPrefix                                    = "batch_"
+	HttpVersion                                      = "HTTP/1.1"
+	HttpNewline                                      = "\r\n"
+	BatchDeleteOperationType  BlobBatchOperationType = "delete"
+	BatchSetTierOperationType BlobBatchOperationType = "set tier"
+)
+
+type BlobBatchBuilder struct {
+	Endpoint    *string
+	AuthPolicy  policy.Policy
+	SubRequests []*policy.Request
+}
+
+// createBatchID is used for creating a new batch id which is used as batch boundary in the request body
+func createBatchID() (string, error) {
+	batchID, err := uuid.New()
+	if err != nil {
+		return "", err
+	}
+
+	return BatchIdPrefix + batchID.String(), nil
+}
+
+// getBatchRequestDelimiter is used for creating the batch boundary
+// e.g. --batch_357de4f7-6d0b-4e02-8cd2-6361411a9525
+// last line of the request body: --batch_357de4f7-6d0b-4e02-8cd2-6361411a9525--
+func getBatchRequestDelimiter(batchID *string, prefixDash bool, postfixDash bool) string {
```

Comment:
Take a look at the [`multipart`](https://pkg.go.dev/mime/multipart) package in the standard library.  You can use it for reading/writing multipart form data.

Reply:
Done